tests/test_args.py

```diff
@@ -171,10 +168,66 @@ def test_invalid_disease_exclusion():
             report_date=today,
             reference_dates=[today - timedelta(days=1), today - timedelta(days=2)],
             data_path="gold/",
-            data_container=None,
+            data_container="test-container",
             production_date=today,
             job_id="test-job-id",
             as_of_date=as_of_date,
             task_exclusions=task_exclusions,
             output_container="test-container",
         )
+
+
+@pytest.mark.parametrize(
+    argnames="raw_val, valid_opts, should_fail",
+    argvalues=[
+        # Test all
+        ("all", all_diseases, False),
+        ("all", nssp_valid_states, False),
+        # Test multiple
+        ("COVID-19, Influenza", all_diseases, False),
+        ("COVID-19,Influenza", all_diseases, False),
+        ("COVID-19,Influenza,RSV", all_diseases, False),
+        ("WA, CA", nssp_valid_states, False),
+        ("WA,CA   , TX", nssp_valid_states, False),
+        ("WA,CA,TX,NY", nssp_valid_states, False),
+        # Test single
+        ("COVID-19", all_diseases, False),
+        ("Influenza", all_diseases, False),
+        ("RSV", all_diseases, False),
+        (" RSV", all_diseases, False),
+        ("COVID-19 ", all_diseases, False),
+        (" COVID-19", all_diseases, False),
+        (" COVID-19 ", all_diseases, False),
+        ("OH", nssp_valid_states, False),
+        ("CA", nssp_valid_states, False),
+        ("WA ", nssp_valid_states, False),
+        (" WA", nssp_valid_states, False),
+        (" WA ", nssp_valid_states, False),
+        # Test failures,
+        ("ZZ,COVID-19", all_diseases, True),
+        ("COVID-19,ZZ", all_diseases, True),
+        ("ZZ,WA", nssp_valid_states, True),
+        ("OH,WA,OO", nssp_valid_states, True),
```

Comment:
```suggestion
        ("OH,WA,OO", nssp_valid_states, True),
        (32, nssp_valid_states, True),
        (["OH","WA","OO"], nssp_valid_states, True)
```

Add some weird types to the call, here a number and a list?

Reply:
I think the list input is important to test here!